### PR TITLE
fix: add proof size check when the VRF proof is empty

### PR DESCRIPTION
**Pull request by 170210**
## Description

This PR added an error handle in proof size check when the VRF proof is empty.

Previously, we allowed the proof hash to be 0 in Validateproof because there was no proof required when verifying Genesis block's information. However, since there is no Genesis block, the proof hash should not be zero in general verification.

I tested it with finschia-sdk, and there is no problem when this handling was added.

related #510

### CODE REVIEW

**Comment by ulbqb** [comment]
https://github.com/Finschia/ostracon/blob/898b48c4c66a9e9a7e148f1b0e61669f3ba197f7/crypto/ed25519/migration.go#L41-L43

It looks like empty proofs are admitted intentionally. Would you research this?

File: crypto/ed25519/migration.go [hunk 1]
```diff
@@ -41,14 +41,15 @@ func ProofToHash(proof []byte) ([]byte, error) {
...
+	proofSize := len(h)
+	if proofSize == 0 {
+		return fmt.Errorf("proof is empty")
+	}
+	if proofSize != voivrf.ProofSize && proofSize != r2vrf.ProofSize {
+		return fmt.Errorf("expected size to be %d bytes, got %d bytes",
+			voivrf.ProofSize,
+			proofSize,
+		)
 	}
```
Comment (ulbqb) [inline crypto/ed25519/migration.go:53]:
```suggestion
	proofSize := len(h)
	if proofSize != voivrf.ProofSize && proofSize != r2vrf.ProofSize {
		return fmt.Errorf("expected size to be %d bytes, got %d bytes",
			voivrf.ProofSize,
			proofSize,
		)
	}
```

Comment (ulbqb) [inline crypto/ed25519/migration.go:53]:
```suggestion
	proofSize := len(h)
	if proofSize != voivrf.ProofSize && proofSize != r2vrf.ProofSize {
		return fmt.Errorf("expected size to be %d bytes, got %d bytes",
			voivrf.ProofSize,
			proofSize,
		)
	}
```

Comment (ulbqb) [reply crypto/ed25519/migration.go:53]: It's ok not to add a special error for empty proof.

**Comment by ulbqb** [comment]
Please add following contents to description:

- Related PRs
- Process where you decided to remove empty proof handling (What you tried)
- The reason where this handling was added

**Comment by ulbqb** [comment]
https://github.com/Finschia/ostracon/blob/ae67c39c8a79827fc0aeeddf86f64bdbf193cbf4/spec/core/data_structures.md?plain=1#L56
And update spec please.

**Review by ulbqb — APPROVED** [review]

**Review by Mdaiki0730 — APPROVED** [review]

**Pull request merged**